### PR TITLE
fix(es/codegen): emit numeric value for Binary and Octal number

### LINKED ISSUE #4870: Binary and Octal literals are not being transpiled to ES5 properly
### Describe the bug

Binary and Octal literals are not being transpiled properly to ES5 syntax. The support for Binary and Octal literals were introduced in ES2015: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#numeric_literals

### Input code

```typescript
function test() {
    return (0b111110111 === 503 && 0o767 === 503);
}
test();
```

### Config

```json
{
  "jsc": {
    "target": "es5",
    "parser": {
      "syntax": "ecmascript"
    }
  },
  "module": {
    "type": "commonjs"
  },
  "sourceMaps": true
}
```

### Playground link

https://play.swc.rs/?version=1.2.189&code=H4sIAAAAAAAAA0srzUsuyczPUyhJLS7R0FSo5lIAgqLUktKiPAUNgyRDEDAAYgVbW1sFUwNjBTU1BYN8czNzmICmNVctF0S3NQD4ftayTgAAAA%3D%3D&config=H4sIAAAAAAAAA0WMSwrDMAxE76K1t934Dj2EcJXg4B%2BSAjXGd69FDF1pNDNvBlwSwA9oyEJsSnpR%2FIIHChklcGwKbtWWpXzTdKDIJ6k15LWiVKvQEzrIscSj206ouTGJgD8wiUVYzkT7nWsn189txgDtbV1Dci2XwPwPbTjKe7cf%2FAesTJxCuQAAAA%3D%3D

### Expected behavior

The numerical literals should be converted to integers. (For the octal literal `0o767` ,  it should be either `0767` or `503`). 

The following output for the input code was obtained from Babel transpiler. 

```typescript
var test = function test() {
  return 503 === 503 && 503 === 503;
};
test();
```

### Actual behavior

The binary and octal literals are not transpiled correctly.

The following output was obtained using swc:

```typescript
var test = function test() {
    return 0b111110111 === 503 && 0o767 === 503;
};
test();
```

The above code when run in ES5 environment throws the following error:

```
missing ; before statement
    <refname> : Line(2) column(24)
    return (0b111110111 === 503 && 0o767 === 503);
```

### Version

1.2.189

### Additional context

_No response_

### LINKED ISSUE #5095: Wrong number literal code from numeric separator
### Describe the bug

swc does not handle code like `1_2_3_4 .toString();`

### Input code

```typescript
1_2_3_4 .toString();

// Note: there is a space between the number and dot`.`
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": false
    },
    "target": "es2022"
  },
  "module": {
    "type": "es6"
  },
  "isModule": true
}
```

### Playground link

[swc playground](https://play.swc.rs/?version=1.2.208&code=H4sIAAAAAAAAAzOMN4o3jjdR0CvJDy4pysxL19C0BgA4YAKfFAAAAA%3D%3D&config=H4sIAAAAAAAAAy2LQQ6AIAwE%2F9IzB8PBA3%2FwEQ1WgxElbUkkhL8LxttmZrbCIR5chYQsxGNJuRQfcEA%2BongOScH0rKMNT6FmQJF30pGInayFjuK95pPGX0uiT82DB1l%2Bo5ypvdeOdBRwAAAA)

### Expected behavior

```JavaScript
1_2_3_4..toString();
```

### Actual behavior

```JavaScript
1_2_3_4.toString();
```

### Version

1.2.208

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
- fix #4870 
- fix #5095

### CODE REVIEW

**Comment by magic-akari** [comment]
I don't know if it's worth adding a pass to handle literals. @kdy1 

Solution B: pass FeatureFlag to codegen pass.

**Comment by kdy1** [comment]
I think this pass is fine, although I prefer codegen because of performance reasons

**Comment by magic-akari** [comment]
> I think this pass is fine, although I prefer codegen because of performance reasons

Ok, I prefer to do it in codegen as well. This is a quick fix. We should pass FeatureFlag into codegen pass in future refactor.

File: crates/swc/tests/tsc-references/propertyAccessNumericLiterals.es6_es5.1.normal.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // @target: es6
 0xffffffff.toString();
-0o01234.toString();
-0b01101101.toString();
+668.toString();
```
Comment (kdy1) [inline crates/swc/tests/tsc-references/propertyAccessNumericLiterals.es6_es5.1.normal.js:3]: This seems like a regression.
This should be `668..toString()`

File: crates/swc_ecma_codegen/src/lib.rs [hunk 2]
```diff
@@ -600,7 +664,15 @@ where
         } else {
             match &v.raw {
                 Some(raw) => {
-                    if self.cfg.target < EsVersion::Es2021 && raw.contains('_') {
+                    if raw.len() > 2 && self.cfg.target < EsVersion::Es2015 && {
```
Comment (magic-akari) [inline crates/swc_ecma_codegen/src/lib.rs:667]: I don't believe any engine supports bigint but not Es2015.
Can we ignore this compatibility? @kdy1

Comment (kdy1) [reply crates/swc_ecma_codegen/src/lib.rs:667]: Yeah I think so

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_codegen

**Pull request merged**